core/src/test/java/org/elasticsearch/indices/memory/IndexingMemoryControllerTests.java

```diff
@@ -139,13 +133,13 @@ public void testShardAdditionAndRemoval() {
                 .put(IndexingMemoryController.INDEX_BUFFER_SIZE_SETTING, "10mb")
                 .put(IndexingMemoryController.TRANSLOG_BUFFER_SIZE_SETTING, "100kb").build());
         final ShardId shard1 = new ShardId("test", 1);
-        controller.setTranslog(shard1, randomInt(10), randomInt(10));
+        controller.simulateIndexing(shard1);
         controller.forceCheck();
         controller.assertBuffers(shard1, new ByteSizeValue(10, ByteSizeUnit.MB), new ByteSizeValue(64, ByteSizeUnit.KB)); // translog is maxed at 64K
 
         // add another shard
         final ShardId shard2 = new ShardId("test", 2);
-        controller.setTranslog(shard2, randomInt(10), randomInt(10));
+        controller.simulateIndexing(shard2);
```

Comment:
I wonder if simulate indexing should also call forceCheck (like real indexing would do)..